specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2020-04-01-preview/keys.json

```diff
@@ -0,0 +1,557 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2020-04-01-preview",
+    "title": "KeyVaultManagementClient",
+    "description": "The Azure management API provides a RESTful set of web services that interact with Azure Key Vault."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/vaults/{vaultName}/keys/{keyName}": {
+      "put": {
+        "tags": [
+          "Keys"
+        ],
+        "operationId": "Keys_CreateIfNotExist",
+        "description": "Creates the first version of a new key if it does not exist. If it already exists, then the existing key is returned without any write operations being performed. This API does not create subsequent versions, and does not update existing keys.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group which contains the specified key vault."
+          },
+          {
+            "name": "vaultName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9-]{3,24}$",
+            "description": "The name of the key vault which contains the key to be created."
+          },
+          {
+            "name": "keyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9-]{1,127}$",
+            "description": "The name of the key to be created."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/KeyCreateParameters"
+            },
+            "description": "The parameters used to create the specified key."
+          }
+        ],
```

Comment:
Tip: rather than duplicating these parameters over and over, you could put them in a file like common.json and reference them across all control plane swaggers like we did for data plane. Also helps avoid unexpected/undeseriable differences.